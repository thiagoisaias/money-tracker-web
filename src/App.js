import React, { Component } from "react";
import { Route, Switch, Redirect } from "react-router-dom";

import Home from "./components/Home";
import Login from "./components/Login";
import Layout from "./components/Layout";
import NotFound from "./components/NotFound";

import TransactionForm from "./components/TransactionForm";

class App extends Component {
  render() {
    return (
      <Switch>
        <Route exact path="/login" component={Login} />
        <Layout>
          <Route exact path="/" component={Home} />
          <Route exact path="/transaction" component={TransactionForm} />
        </Layout>
      </Switch>
    );
  }
}

export default App;
