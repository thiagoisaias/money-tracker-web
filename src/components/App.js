import React, { Component } from "react";
import { Route, Switch } from "react-router-dom";
import Home from "./Home";
import Layout from "./Layout";
import Login from "./Auth/Login";
import TransactionForm from "./TransactionForm";

class App extends Component {
  render() {
    return (
      <Switch>
        <Route exact path="/login" component={Login} />
        <Layout>
          <Route exact path="/" component={Home} />
          <Route exact path="/transaction" component={TransactionForm} />
        </Layout>
      </Switch>
    );
  }
}

export default App;
