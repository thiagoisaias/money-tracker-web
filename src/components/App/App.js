import React, { Component } from "react";
import { Route, Switch } from "react-router-dom";

import Home from "../Home/Home";
import SignupPage from "../SignupPage/SignupPage";
import TransactionForm from "../TransactionForm/TransactionForm";

class App extends Component {
  render() {
    return (
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/transaction" component={TransactionForm} />
        <Route component={SignupPage} />
      </Switch>
    );
  }
}

export default App;
