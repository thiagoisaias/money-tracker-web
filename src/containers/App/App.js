import React from "react";
import { Route, Switch } from "react-router-dom";

import AccountFormContainer from "../Accounts/AccountForm/AccountFormContainer";
import AccountListContainer from "../Accounts/AccountList/AccountListContainer";
import Home from "../../components/Home/Home";
import SignupPage from "../../components/SignupPage/SignupPage";
import TransactionForm from "../../components/Transactions/TransactionForm/TransactionForm";

const App = props => {
  return (
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/auth" component={SignupPage} />
      <Route exact path="/transaction" component={TransactionForm} />
      <Route exact path="/accounts" component={AccountListContainer} />
      <Route exact path="/accounts/new" component={AccountFormContainer} />
    </Switch>
  );
};

export default App;
