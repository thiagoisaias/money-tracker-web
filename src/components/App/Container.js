import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import App from "./App";

export class Container extends Component {
  static propTypes = {
    isAuthenticated: PropTypes.bool.isRequired
  };

  render() {
    return <App isAuthenticated={this.props.isAuthenticated} />;
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
});

export default withRouter(connect(mapStateToProps)(Container));
