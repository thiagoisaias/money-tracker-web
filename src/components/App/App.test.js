import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import App from "./App";

describe("App", () => {
  const mockProps = {
    isAuthenticated: true
  };
  const component = shallow(<App {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
