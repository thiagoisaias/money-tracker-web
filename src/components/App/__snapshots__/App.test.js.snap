// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/transaction"
  />
  <Route
    component={[Function]}
  />
</Switch>
`;
